src/test/java/org/jenkinsci/plugins/workflow/steps/RetryStepTest.java

```diff
@@ -179,4 +179,50 @@ public void downstreamBuildFailureShouldRetry() throws Exception {
         r.assertLogContains("trying 2", b);
         r.assertLogContains("trying 3", b);
     }
+
+    @Issue("JENKINS-44379")
+    @Test
+    public void shouldNotRetryAfterOuterTimeout() throws Exception {
+        WorkflowJob p = r.createProject(WorkflowJob.class);
+        p.setDefinition(
+                new CpsFlowDefinition(
+                        "int count = 0\n"
+                                + "timeout(time: 5, unit: 'SECONDS') {\n"
```

Comment:
nit: The indentation here and in the next test is weird. I prefer to use trailing `+` operators for Pipeline scripts in tests so that the script all starts at the same level of indentation.

Reply:
I started formatting my code with `google-java-format` a long time ago in order to avoid having to debate formatting preferences.